docs/apis/es_sender.md

```diff
@@ -0,0 +1,121 @@
+# elasticsearch_sender_api
+
+The `elasticsearch_sender_api` will provide a factory that can create sender apis that can be accessed in any operation through the `getAPI` method on the operation.
+
+
+This is a [Factory API](https://terascope.github.io/teraslice/docs/packages/job-components/api/interfaces/apifactoryregistry), which can be used to fully manage api creation and configuration.
+
+
+## Elasticsearch Sender Factory API Methods
+
+### size
+
+this will return how many seperate sender apis are in the cache
+
+### get
+parameters:
+- name: String
+
+this will fetch any sender api that is associated with the name provided
+
+### getConfig
+parameters:
+- name: String
+
+this will fetch any sender api config that is associated with the name provided
+
+### create (async)
+parameters:
+- name: String
+- configOverrides: Check options below, optional
+
+this will create an instance of a sender api, and cache it with the name given. Any config provided in the second argument will override what is specified in the apiConfig and cache it with the name provided. It will throw an error if you try creating another api with the same name parameter
+
+```typescript
+    const apiManager = this.getAPI<ElasticSenderFactoryAPI>(apiName);
+    // this will return an api cached at "normalClient" and this instance will use all configurations listed on the apiConfig
+    const client = apiManager.create('normalClient')
+
+    // this will return an api cached at "overrideClient" and this instance will have an override setting the parameter index to "other_index", this will use the rest of the configurations listed in the apiConfig
+    const overrideClient = apiManager.create('overrideClient', { index: 'other_index'})
```

Comment:
It is root default indentation here, also the code comments are kind of confusing

```suggestion
const apiManager = this.getAPI<ElasticSenderFactoryAPI>(apiName);
// this will return an api cached at "normalClient" and it will use the default api config
const client = apiManager.create('normalClient');

// this will return an api cached at "overrideClient" and it will use the api config but override the index to "other_index" in the new instance.
const overrideClient = apiManager.create('overrideClient', { index: 'other_index' });
```